### DISCUSSION TITLE
Additional dependencies

### DISCUSSION THREAD

**Discussion by abhi1693** [cdalvaro/docker-salt-master#233] [Q&A] [answered]
What is the best way to install additional dependencies into the container so that it persists across reboots?

**Comment by cdalvaro** [comment]
Hi @abhi1693, thank you for your question!

Please, could you provide more information about this? Do you mean dependencies such as Python packages?

**Reply by abhi1693** [reply]
I wanted to explore the redis cache with the master and came across https://docs.saltproject.io/en/latest/ref/cache/all/salt.cache.redis_cache.html it suggests to run pip install for redis package. Now, I generally understand the requirements but I was wondering if I could I install this dependency without losing it when I reboot the container or upgrade it. If it's required I can also create my custom docker image on top of yours.

**Reply by cdalvaro** [reply]
This is something I've been thinking about sometimes, but I've never had the necessity.

This seems to be a good reason to add support for installing extra dependencies. I'll add this support with an image revision in the following days.

I think this can be implemented by providing a requirements.txt file or using an environment variable with packages to install.

Do you have any suggestion?

**Reply by cdalvaro** [reply, marked as answer]
A new image has been release: `ghcr.io/cdalvaro/docker-salt-master:3007.0_1`

If you find any issue with this release, please, let me know and feel free to reopen issue #234
